File: tests/fixtures/real_world/typescript/cfg/error_handling.ts
import * as fs from 'fs';

function readConfigUnsafe(path: string): string {
    var content = fs.readFileSync(path, 'utf8');
    var config = JSON.parse(content);
    if (config.error) {
        console.log('Error in config');
        // falls through without returning!
    }
    return config.value;
}

function readConfigSafe(path: string): string {
    var content = fs.readFileSync(path, 'utf8');
    var config = JSON.parse(content);
    if (config.error) {
        throw new Error('Invalid config: ' + config.error);
    }
    return config.value;
}

File: tests/fixtures/real_world/rust/cfg/error_handling.rs
use std::fs;
use std::io;

fn read_config(path: &str) -> Result<String, io::Error> {
    let content = fs::read_to_string(path)?;
    Ok(content)
}

fn read_config_panicky(path: &str) -> String {
    fs::read_to_string(path).unwrap()
}

fn read_config_expect(path: &str) -> String {
    fs::read_to_string(path).expect("config must exist")
}
